src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -701,8 +704,21 @@ string ReserveName(string prefix)
                 EmitUncaptureUntil("0");
             }
 
-            // We're done.  Patch up any additional declarations.
+            // We're done with the match.
+
+            // Patch up any additional declarations.
             ReplaceAdditionalDeclarations(writer, additionalDeclarations, additionalDeclarationsPosition, additionalDeclarationsIndent);
+
+            // And emit any required helpers.
+            foreach (KeyValuePair<string, string[]> localFunctions in additionalLocalFunctions.OrderBy(k => k.Key))
```

Comment:
NIT: in terms of readability, should we emit a `return;` statement before the local functions so that it is clear that they are not just part of the regular method execution?

Reply:
Or perhaps just a comment in case having return; statement isn't a style adopted by most people

Reply:
yeah sounds good to me.